### ISSUE TITLE
Feature request: Support webassembly (WASM)

### ISSUE THREAD

**Issue by sanathusk** [pocketbase/pocketbase#1839]
As a user of Pocket Base, I would like to request support for WebAssembly (WASM) for several reasons:
- **Access to a Larger Community:** WASM has a large and growing community of developers and libraries, which would open up new opportunities for Pocket Base to integrate with other tools and technologies.
- **Better Compatibility:** WebAssembly is supported by all modern browsers, making it easier for Pocket Base to reach a wider audience. This also eliminates the need to worry about compatibility issues across different browsers and platforms.
- **Increased Security:** WebAssembly code is executed in a secure sandbox environment that is separate from the main JavaScript runtime. This improves security and protects users against malicious attacks and exploits.

In conclusion, supporting WebAssembly in Pocket Base would bring numerous benefits, access to a larger community, better compatibility, and increased security. These benefits would help Pocket Base stay ahead of the curve by attracting more developers/users.

**Comment by ganigeorgiev** [comment]
Sorry, there are no plans for integrated wasm support because it will be hard to maintain, test and document.
You can find a similar discussion in https://github.com/pocketbase/pocketbase/discussions/410.

Support for JS app event hooks via [goja](https://github.com/dop251/goja) will be added in the near future.

**Comment by nojaf** [comment]
Out of interest, can discussion 410 still be read somewhere?

**Comment by ganigeorgiev** [comment]
@nojaf I'm not sure. Based on partial email notifications the original author seems to be gedw99 but their GitHub account is no longer accessible which probably was propagated to all of their content too.

In all cases, at least for now WASM support remains out of the v1 scope as it would require to create and maintain a completely different set of APIs. Note that users are free to work on it if they want to as it doesn't need to be builtin and can be implemented as external and optional "plugin" similar to the [goja JSVM](https://github.com/pocketbase/pocketbase/tree/master/plugins/jsvm).

**Comment by nojaf** [comment]
Thanks! That makes sense to keep it out of scope. I was just wondering if there are any limitations on a technical level. I might explore this at some point, but I'm in no rush.
